cmd/agent/main.go

```diff
@@ -179,24 +178,26 @@ func main() {
 	// !Security
 
 	// Edge
-	edgeManagerParameters := &edge.ManagerParameters{
-		Options:           options,
-		AdvertiseAddr:     advertiseAddr,
-		ClusterService:    clusterService,
-		DockerInfoService: dockerInfoService,
-		ContainerPlatform: containerPlatform,
-	}
-	edgeManager := edge.NewManager(edgeManagerParameters)
-
+	var edgeManager *edge.Manager
 	if options.EdgeMode {
-		edgeKey, err := retrieveEdgeKey(options.EdgeKey, clusterService)
+		edgeManagerParameters := &edge.ManagerParameters{
+			Options:           options,
+			AdvertiseAddr:     advertiseAddr,
+			ClusterService:    clusterService,
+			DockerInfoService: dockerInfoService,
+			ContainerPlatform: containerPlatform,
+		}
+		edgeManager = edge.NewManager(edgeManagerParameters)
+
+		edgeKey, err := edge.RetrieveEdgeKey(options.EdgeKey, clusterService)
 		if err != nil {
 			log.Printf("[ERROR] [main,edge] [message: Unable to retrieve Edge key] [error: %s]", err)
 		}
 
 		if edgeKey != "" {
 			log.Println("[DEBUG] [main,edge] [message: Edge key found in environment. Associating Edge key]")
 
+			// TODO: this fails when RetreiveEdgeKey came from an existing file, as mode 0444 can't be written to - unless you're root...
```

Comment:
so will this fail if the user restarts the process? as a file is saved on the first time edgeKey is provided. maybe we should separate methods, or change the signature to `SetKey(edgeKey, fromFile)`

Reply:
firstly, i didn't want to change what the code does now - but tbh, i'm not sure that having an agent change its key without the user explicitly asking is a good idea? @jamescarppe and @deviantony have you any thoughts if we've ever told anyone about changing the edgekey on an existing agent?

Reply:
I don't think that's a use case we've ever considered/heard about.

Reply:
@SvenDowideit you could do one of the following things:

1. An atomic write: write to a temp file and move over the existing one
2. Chmod +w, write, chmod -x

Reply:
@deviantony well for dev purposes, it's useful, but I don't think it's necessary

Reply:
Should this be considered as part of a broader "running the agent as non-root" PR ? Related to my other comment above.